### PR TITLE
Added usage of aws session token to use temporary credentials.

### PR DESCRIPTION
**Pull request by salyarka**
Allows to use temporary aws credentials with session token.

### CODE REVIEW

File: tests/test_settings.py [hunk 1]
```diff
@@ -20,6 +20,24 @@ def test_ses_access_key_given(self):
         self.assertEqual(django_ses.settings.ACCESS_KEY, settings.AWS_SES_ACCESS_KEY_ID)
         self.assertEqual(django_ses.settings.SECRET_KEY, settings.AWS_SES_SECRET_ACCESS_KEY)
 
+    def test_aws_session_token_given(self):
+        settings.AWS_SESSION_TOKEN = "FwoGZXIvYXdzED8aDAILqEtZvcDCx+KsFCK1AUwcLbm4d+mAlRWYN+r1adKoIfwe/T117KNcql" \
+            "fbFFc6lgM1BQk9RepOZOyhNnx1ji12BMnA+Sc/9H1gi/QRt51U0EQVhcT7i9YZbipzrYMLpvxe0dwXwC7MTy7NQRkEMhpyXWgFw4Wz+" \
+            "pHdZTFI4DOEhjf/t1FcuV2jX0oS0Eqqck2YB6yY03FpQRFVFIKUFcyvt9kMP9F77iHkgnEWBxOVcfSxBHfgQDTCHCecMNDN02/u628o" \
+            "xK6elAYyLZu54kuwLAbe3hD2++FpbjCSF88DFWESks8o2PP489XCCCJrX/SnurGNfeWifA=="
```
Comment (pcraciunoiu) [inline tests/test_settings.py:27]: For testing, a shorter, fake value would be sufficient instead of these very long strings.

**Review by pcraciunoiu — COMMENTED** [review]
What is the use case for this? I'm trying to understand how useful it would be to add it.

**Comment by salyarka** [comment]
Use case for example: i want to use django-ses with temporary credenttials that i got from [AssumeRole](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html) operation, and i cant do this, cause [boto3.client](https://github.com/django-ses/django-ses/blob/master/django_ses/__init__.py#L94) call dont accepts session token that is required for using temporary credentials (to use temporary credentials you must provide aws_access_key_id, aws_secret_access_key and aws_session_token to boto3). This fork allows to use temporary credentials.

File: django_ses/__init__.py [hunk 2]
```diff
@@ -95,6 +96,7 @@ def open(self):
                 'ses',
                 aws_access_key_id=self._access_key_id,
                 aws_secret_access_key=self._access_key,
+                aws_session_token=self._session_token,
```
Comment (pcraciunoiu) [inline django_ses/__init__.py:99]: Should this be added to all places where `boto3` client or connections are used? for example in `ses_email_address.py`, `get_ses_statistics.py`, [`dashboard`](https://github.com/django-ses/django-ses/blob/e64606125e7a56900d44f1e4169a991ec4afbbe8/django_ses/views.py#L98)

**Review by pcraciunoiu — COMMENTED** [review]
OK the use case makes sense. Could you update the README to document this usage as well?

File: django_ses/__init__.py (hunk 2 above)
Comment (salyarka) [reply django_ses/__init__.py:99]: Yes, it should be added, soon i will make changes to all boto3 client initialization and to README.

File: README.rst [hunk 3]
```diff
@@ -463,6 +463,12 @@ Full List of Settings
   *Required.* Alternative API keys for Amazon SES. This is useful in situations
   where you would like to use separate access keys for different AWS services.
 
+``AWS_SES_SESSION_TOKEN``, ``AWS_SES_SECRET_ACCESS_KEY``
+  Optional. Use one of this variables, to provide session token that is required
+  when temporary credentials are used. Details:
```
Comment (pcraciunoiu) [inline README.rst:468]:
```suggestion
  Optional. Use `AWS_SES_SESSION_TOKEN` to provide session token
  when temporary credentials are used. Details:
```

**Review by pcraciunoiu — APPROVED** [review]

**Comment by pcraciunoiu** [comment]
I published this on PyPi as 3.1.0, but for some reason it's not picking it up. Maybe they have some issues. You can check later, if you want.

Thanks for your contribution!

**Comment by pcraciunoiu** [comment]
I'm not 100% sure that you are right, but it sounds right :) And so if you want to PR an approach that is more elegant than option 2, I'd take a look at it.

I'm not actively using this on a project now, so it's tricky for me to test it, and I would trust that you'd be able to test it in a real environment for a bit before we merge it.

**Comment by colehorvitz** [comment]
@pcraciunoiu After looking at the way django-storages handles this, it seems they expose an AWS_S3_SESSION_PROFILE config. [AWS profiles support temporary credentials out of the box](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-role.html#cli-role-overview), so my instinct is to think that doing something similar to django-storages would be the best. That is, provide an alternative means of authenticating via AWS_SES_SESSION_PROFILE.

I'm not a boto/AWS expert in general, but I'd be happy to take a crack at this.

**Comment by pcraciunoiu** [comment]
@colehorvitz that sounds reasonable to me, I see it in use here and should be fairly straightforward
https://github.com/jschneier/django-storages/blob/70fa8d96665f0c5e44aacd6c782f6489fa8dc11e/storages/backends/s3.py#L313

I'm not an expert either, and there seem to be many ways to specify credentials!

**Pull request merged**